src/services/injectableOptions.ts

```diff
@@ -1,290 +1,177 @@
 import { FormTypes, SubmissionTypes } from '@oneblink/types'
 import { formElementsService, typeCastService } from '@oneblink/sdk-core'
-
 import { localisationService } from '@oneblink/apps'
 import { v4 as uuidv4 } from 'uuid'
-import { TaskContext } from '../hooks/useTaskContext'
 import { MiscTypes } from '@oneblink/types'
+import { TaskContext } from '../hooks/useTaskContext'
+
+type Option = Pick<FormTypes.ChoiceElementOption, 'value' | 'label'>
+
+type Match = Map<string, Match | undefined>
+
+function groupElementNames(memo: Match, rootElementNames: string[]): Match {
```

Comment:
why does this bother recurse all the way down when processInjectableDynamicOption will just run it again when it goes down the next layer?

Reply:
We need to get the full tree to pass the subtree down recursively

Reply:
Or maybe we don't...Let me try some tests with a short circuit.